diff -Nru autospec-1.4.7/autospec.conf.in autospec-1.4.7.patched/autospec.conf.in
--- autospec-1.4.7/autospec.conf.in	2009-06-28 20:18:14.000000000 +0200
+++ autospec-1.4.7.patched/autospec.conf.in	2009-07-08 21:15:25.186969362 +0200
@@ -537,4 +537,5 @@
 rpms_dir="`rpm --eval %_rpmdir 2>/dev/null`"
 spec_dir="`rpm --eval %_specdir 2>/dev/null`"
 source_dir="`rpm --eval %_sourcedir 2>/dev/null`"
+build_dir="`rpm --eval %_builddir 2>/dev/null`"
 tmppath_dir="`rpm --eval %_tmppath 2>/dev/null`"
