diff -Nru autospec-1.4.17.orig//lib/librepository.lib.in autospec-1.4.17/lib/librepository.lib.in
--- autospec-1.4.17.orig//lib/librepository.lib.in	2009-06-28 21:44:48.000000000 +0200
+++ autospec-1.4.17/lib/librepository.lib.in	2010-07-01 03:46:25.369016704 +0200
@@ -361,15 +361,15 @@
                      sed -n "\
                         /<a href=\"/{s|.*<a href=\"\([^\"]*\)\".*|\1|;bm;};
                         /<A HREF=\"/{s|.*<A HREF=\"\([^\"]*\)\".*|\1|;bm;};
-                        /^$pck_name-\(svn\)\?[0-9]\+/bm;be;
-                        :m;/^$pck_name-\(svn\)\?[0-9]\+/{s,.*,$curr_arch:&,p}
+                        /^$pck_name-[^-]*-[^-]*\.rpm/bm;be;
+                        :m;/^$pck_name-[^-]*-[^-]*\.rpm/{s,.*,$curr_arch:&,p}
                         :e" $flist | grep -v -- "$excludeopt"
                   else
                      sed -n "\
                         /<a href=\"/{s|.*<a href=\"\([^\"]*\)\".*|\1|;bm;};
                         /<A HREF=\"/{s|.*<A HREF=\"\([^\"]*\)\".*|\1|;bm;};
-                        /^$pck_name-\(svn\)\?[0-9]\+/bm;be;
-                        :m;/^$pck_name-\(svn\)\?[0-9]\+/{s,.*,$curr_arch:&,p}
+                        /^$pck_name-[^-]*-[^-]*\.rpm/bm;be;
+                        :m;/^$pck_name-[^-]*-[^-]*\.rpm/{s,.*,$curr_arch:&,p}
                         :e" $flist
                   fi ))
          done
