diff -Nru autospec-1.4.9.orig/templates/python autospec-1.4.9/templates/python
--- autospec-1.4.9.orig/templates/python	2008-05-11 22:41:59.000000000 +0200
+++ autospec-1.4.9/templates/python	2010-01-06 10:55:10.781675316 +0100
@@ -22,11 +22,11 @@
 %setup -q
 
 %build
-CFLAGS=\"%{optflags}\" python setup.py build
+CFLAGS=\"%{optflags}\" %{__python} setup.py build
 
 %install
 [ "%{buildroot}" != / ] && rm -rf "%{buildroot}"
-python setup.py install \
+%{__python} setup.py install \
    --root="%{buildroot}" \
    --install-headers=%{_includedir}/python \
    --install-lib=%{python_sitearch}
