--- libapse.lib.orig    2008-10-10 01:39:44.000000000 +0200
+++ libapse.lib 2008-10-10 01:50:06.000000000 +0200
@@ -429,7 +429,8 @@
    unset APSE_LAST_VERSION

    if [ "$last_version" ]; then
-      APSE_LAST_VERSION="$last_version"
+      last_version="${last_version/-/.}"
+      APSE_LAST_VERSION="${last_version}"
       notify.note $"\
 last version of \`${NOTE}$specname${NORM}' found:"" \
 \`${NOTE}$last_version${NORM}'"
