--- autospec/plugins/pck-update.in	2010-05-12 22:12:11.000000000 +0200
+++ autospec/plugins/pck-update	2010-05-25 16:42:42.000000000 +0200
@@ -2040,10 +2040,8 @@
 
          local f rpm_provided_by buildrequirements_to_install
          for f in ${SPEC_BUILDREQUIRES[@]}; do
-            unset rpm_provided_by
-            rpm --quiet -q $rpm_root_opts --whatprovides $f &>/dev/null
-            [ $? -eq 0 ] &&
-               rpm_provided_by="$(rpm -q $rpm_root_opts --whatprovides $f)"
+            rpm_provided_by="$(rpm -q $rpm_root_opts --whatprovides $f)"
+            [ $? -eq 0 ] || unset rpm_provided_by
 
             notify.note " * "$"\
 build requirement found"": \`${NOTE}$f${NORM}' ""\
