diff -Nru autospec-1.5.0.orig//plugins/pck-update.in autospec-1.5.0/plugins/pck-update.in
--- autospec-1.5.0.orig//plugins/pck-update.in	2010-07-17 23:00:50.000000000 +0200
+++ autospec-1.5.0/plugins/pck-update.in	2010-07-31 19:55:55.222342783 +0200
@@ -1235,7 +1235,7 @@
                   $spec_dir/${SRPM_SPECFILE}.0.old
          fi
 
-         rpm -hiv \
+         rpm -hiv --nodeps \
              --define "%_specdir $spec_dir" \
              --define "%_sourcedir $source_dir" \
              $srpms_dir/$got_SPEC_FILENAME 2>&1 >/dev/null ||
