diff -Nru autospec-1.5.0.orig//autospec.spec.in autospec-1.5.0/autospec.spec.in
--- autospec-1.5.0.orig//autospec.spec.in	2010-07-11 20:58:17.000000000 +0200
+++ autospec-1.5.0/autospec.spec.in	2010-08-09 00:38:28.364052465 +0200
@@ -13,7 +13,7 @@
 Packager:      Davide Madrisan <davide.madrisan@gmail.com>
 URL:           http://www.openmamba.org/packages.html
 Source:        autospec-%{version}.tar.bz2
-PreReq:        rpm
+Requires(pre): rpm
 Requires:      cpio, coreutils, curl, findutils, grep, sed
 Requires:      /bin/mktemp
 Requires:      /usr/bin/getopt
diff -Nru autospec-1.5.0.orig//plugins/pck-update.in autospec-1.5.0/plugins/pck-update.in
--- autospec-1.5.0.orig//plugins/pck-update.in	2010-08-09 00:37:02.782052085 +0200
+++ autospec-1.5.0/plugins/pck-update.in	2010-08-09 00:39:07.550052253 +0200
@@ -758,7 +758,7 @@
                notify.debug "rpm_macro_uninstallinfo check failed"; }
          fi
          if [ "$rpm_macro_installinfo_binary" ]; then
-            grep -q "PreReq[ \t]*:[ \t]*${rpm_macro_installinfo_binary}" \
+            grep -q "Requires(post)[ \t]*:[ \t]*${rpm_macro_installinfo_binary}" \
                $specfile ||
              { let "errors += 1"
                notify.debug "rpm_macro_installinfo_binary check failed"; }
@@ -769,8 +769,8 @@
 ---------------------------------------
 ${NOTE}"$"Hint"":${NORM}
 $([[ "$rpm_macro_installinfo_binary" ]] &&
-   echo "PreReq: $rpm_macro_installinfo_binary" ||
-   echo "PreReq: ${path_installinfo:-/sbin/install-info}")
+   echo "Requires(post):$rpm_macro_installinfo_binary" ||
+   echo "Requires(post):${path_installinfo:-/sbin/install-info}")
 
 %post [<subpackage>]
 $([[ "$rpm_macro_installinfo" ]] &&
diff -Nru autospec-1.5.0.orig//plugins/spec-create.in autospec-1.5.0/plugins/spec-create.in
--- autospec-1.5.0.orig//plugins/spec-create.in	2010-04-12 20:04:46.000000000 +0200
+++ autospec-1.5.0/plugins/spec-create.in	2010-08-09 00:39:27.935677047 +0200
@@ -782,8 +782,8 @@
 License: ${spec_license:-$license_defvalue}"
 if [ "$(specfile.infopages --exist)" = 1 ]; then
    [ "$rpm_macro_installinfo_binary" ] &&
-      echo "PreReq: ${rpm_macro_installinfo_binary}" ||
-      echo "PreReq: ${path_installinfo:-/sbin/install-info}"
+      echo "Requires(post):${rpm_macro_installinfo_binary}" ||
+      echo "Requires(post):${path_installinfo:-/sbin/install-info}"
 fi
 [ "$spec_requires" ] && echo Requires: $spec_requires
 [ "$spec_buildrequires" ] && echo BuildRequires: $spec_buildrequires
diff -Nru autospec-1.5.0.orig//tests/test01_pkgquality.in autospec-1.5.0/tests/test01_pkgquality.in
--- autospec-1.5.0.orig//tests/test01_pkgquality.in	2010-05-14 21:33:13.000000000 +0200
+++ autospec-1.5.0/tests/test01_pkgquality.in	2010-08-09 00:40:03.792692378 +0200
@@ -270,8 +270,8 @@
 ---------------------------------------
 ${NOTE}"$"Hint"":${NORM}
 $([[ "$rpm_macro_installinfo_binary" ]] &&
-   echo "PreReq: $rpm_macro_installinfo_binary" ||
-   echo "PreReq: ${path_installinfo:-/sbin/install-info}")
+   echo "Requires(post):$rpm_macro_installinfo_binary" ||
+   echo "Requires(post):${path_installinfo:-/sbin/install-info}")
 
 %post [<subpackage>]
 $([[ "$rpm_macro_installinfo" ]] &&
diff -Nru autospec-1.5.0.orig//unmaintained/mbrowse.spec autospec-1.5.0/unmaintained/mbrowse.spec
--- autospec-1.5.0.orig//unmaintained/mbrowse.spec	2006-11-24 18:58:24.000000000 +0100
+++ autospec-1.5.0/unmaintained/mbrowse.spec	2010-08-09 00:40:35.354052403 +0200
@@ -59,7 +59,7 @@
 #Patch3:        http://www.awebsiteintheworld.org/awebpatch-%{version}.patch
 BuildRequires: net-snmp-devel >= 4.2
 BuildRequires: libopenssl-devel
-PreReq:        %{__install_info}
+Requires(post):%{__install_info}
 %if %with_gui
 BuildRequires: libgtk1-devel >= 1.2.10
 #BuildRequires: libglib2-devel >= 2.1.5
