diff -Nru autospec-1.5.0.orig//plugins/pck-update.in autospec-1.5.0/plugins/pck-update.in
--- autospec-1.5.0.orig//plugins/pck-update.in	2010-08-04 13:53:29.024307664 +0200
+++ autospec-1.5.0/plugins/pck-update.in	2010-08-04 13:56:16.759307837 +0200
@@ -2992,6 +2992,7 @@
 
          # print the old SRPMS package...
          [[ -n "$SPEC_FILENAME_old" && \
+            "$nosrpm" != "1" && \
             "$SPEC_VERSION_old-$SPEC_RELEASE_old" != \
             "$SPEC_VERSION-$SPEC_RELEASE" ]] &&
           { let "srpm_pcks_to_backup = 1"
