diff -Nru autospec-1.5.0.orig//plugins/pck-update.in autospec-1.5.0/plugins/pck-update.in
--- autospec-1.5.0.orig//plugins/pck-update.in	2010-08-08 17:41:33.864557659 +0200
+++ autospec-1.5.0/plugins/pck-update.in	2010-08-08 17:43:53.356307747 +0200
@@ -880,11 +880,11 @@
 function specfile.rotate_bck() {
    local v
    for v in `seq $((${spec_backup_numbers:-5} - 2)) -1 0`; do
-      [[ -e $spec_dir/${SRPM_SPECFILE}.$v.old ]] &&
+      [[ -e $spec_dir/.${SRPM_SPECFILE}.$v.old ]] &&
        { notify.debug "\
-[$spec_dir] ${SRPM_SPECFILE}.$v.old --> ${SRPM_SPECFILE}.$(($v + 1)).old"
-         mv -f $spec_dir/${SRPM_SPECFILE}.$v.old \
-               $spec_dir/${SRPM_SPECFILE}.$(($v + 1)).old; }
+[$spec_dir] .${SRPM_SPECFILE}.$v.old --> .${SRPM_SPECFILE}.$(($v + 1)).old"
+         mv -f $spec_dir/.${SRPM_SPECFILE}.$v.old \
+               $spec_dir/.${SRPM_SPECFILE}.$(($v + 1)).old; }
    done
 }
 
@@ -1230,9 +1230,9 @@
          if [ -e "$spec_dir/$SRPM_SPECFILE" ]; then
             notify.note $"\
 making a ${NOTE}backup copy${NORM} of the current specfile""
-   --> ${NOTE}$spec_dir/${SRPM_SPECFILE}.0.old${NORM}"
+   --> ${NOTE}$spec_dir/.${SRPM_SPECFILE}.0.old${NORM}"
             cp -f $spec_dir/$SRPM_SPECFILE \
-                  $spec_dir/${SRPM_SPECFILE}.0.old
+                  $spec_dir/.${SRPM_SPECFILE}.0.old
          fi
 
          rpm -hiv --nodeps \
@@ -1372,9 +1372,9 @@
 #
 #         notify.note $"\
 #making a ${NOTE}backup copy${NORM} of the current specfile""
-#   --> ${NOTE}$spec_dir/${SRPM_SPECFILE}.0.old${NORM}"
+#   --> ${NOTE}$spec_dir/.${SRPM_SPECFILE}.0.old${NORM}"
 #         cp -f $SRPM_SPECFILE_WITH_PATH \
-#              ${SRPM_SPECFILE_WITH_PATH}.0.old
+#              .${SRPM_SPECFILE_WITH_PATH}.0.old
 
          notify.note $"${NOTE}updating${NORM} the new specfile \
 \`${NOTE}$spec_dir/$SRPM_SPECFILE${NORM}'""..."
