diff -Nru autospec-1.4.17.orig//lib/libapse.lib.in autospec-1.4.17/lib/libapse.lib.in
--- autospec-1.4.17.orig//lib/libapse.lib.in	2010-04-12 23:16:34.000000000 +0200
+++ autospec-1.4.17/lib/libapse.lib.in	2010-06-03 03:02:59.489016996 +0200
@@ -313,7 +313,7 @@
       notify.note $"looking at"" <${NOTE}$curr_curl_url${NORM}> (\`${NOTE}source0${NORM}')..."
 
       notify.debug "curl \$curr_curl_proxy_opts -s -L \"$curr_curl_url\""
-      curl $curr_curl_proxy_opts -s -L "$curr_curl_url" > $fcurlout
+      curl $curl_opts_netlink $curr_curl_proxy_opts -s -L "$curr_curl_url" > $fcurlout
       let "retval = $?"
 
       case "$retval" in
