diff -Nru autospec-1.5.0.orig//plugins/pck-update.in autospec-1.5.0/plugins/pck-update.in
--- autospec-1.5.0.orig//plugins/pck-update.in	2010-08-16 14:02:21.485570021 +0200
+++ autospec-1.5.0/plugins/pck-update.in	2010-08-16 14:03:53.353694250 +0200
@@ -2269,8 +2269,13 @@
          for pck in ${SPEC_TARGET[*]}; do
             # FIXME : we should check and complain if the package has
             # been built for a different architecture
-            filename="$rpms_dir/$target_cpu/\
+            if [ "${SPEC_BUILDARCH:-$BUILD_ARCH}" == "noarch" ]; then
+               filename="$rpms_dir/noarch/\
+$pck-$pck_rpmversion-$pck_rpmrelease$DISTRO_rpm.noarch.rpm"
+            else
+               filename="$rpms_dir/$target_cpu/\
 $pck-$pck_rpmversion-$pck_rpmrelease$DISTRO_rpm.$target_cpu.rpm"
+            fi
             #notify.note " * $filename"
             [ -e $filename ] ||
                notify.error $"package not found"": \`$filename'"
@@ -2283,8 +2288,13 @@
          for pck in ${SPEC_TARGET[*]}; do
             # FIXME : we should check and complain if the package has
             # been built for a different architecture
-            filename="$rpms_dir/$target_cpu/\
+            if [ "${SPEC_BUILDARCH:-$BUILD_ARCH}" == "noarch" ]; then
+               filename="$rpms_dir/noarch/\
+$pck-$pck_rpmversion-$pck_rpmrelease$DISTRO_rpm.noarch.rpm"
+            else
+               filename="$rpms_dir/$target_cpu/\
 $pck-$pck_rpmversion-$pck_rpmrelease$DISTRO_rpm.$target_cpu.rpm"
+            fi
             notify.note " * $filename"
             #[ -e $filename ] ||
             #   notify.error $"package not found"": \`$filename'"
@@ -2389,9 +2399,14 @@
          for pck in ${SPEC_TARGET[*]}; do
             # FIXME : we should check and complain if the package has
             # been built for a different architecture
-            filename="$rpms_dir/$target_cpu/\
+            if [ "${SPEC_BUILDARCH:-$BUILD_ARCH}" == "noarch" ]; then
+               filename="$rpms_dir/noarch/\
+$pck-$pck_rpmversion-$pck_rpmrelease$DISTRO_rpm.noarch.rpm"
+            else
+               filename="$rpms_dir/$target_cpu/\
 $pck-$pck_rpmversion-$pck_rpmrelease$DISTRO_rpm.$target_cpu.rpm"
-             notify.note " * $filename"
+            fi
+            notify.note " * $filename"
             [ -e $filename ] ||
                notify.error $"package not found"": \`$filename'"
          done
@@ -2578,8 +2593,13 @@
             $tool -b $filename | sed "s,*.*/\(.*\),\1,"
             local i=0
             for pck in ${SPEC_TARGET[@]}; do
-               filename="$rpms_dir/$target_cpu/\
+               if [ "${SPEC_BUILDARCH:-$BUILD_ARCH}" == "noarch" ]; then
+                  filename="$rpms_dir/noarch/\
+$pck-$pck_rpmversion-$pck_rpmrelease$DISTRO_rpm.noarch.rpm"
+               else
+                  filename="$rpms_dir/$target_cpu/\
 $pck-$pck_rpmversion-$pck_rpmrelease$DISTRO_rpm.$target_cpu.rpm"
+               fi
                [ -e "$filename" ] ||
                   notify.error $"package not found"": \`$filename'"
                $tool -b $filename | sed "s,*.*/\(.*\),\1,"
@@ -3116,8 +3136,13 @@
          if [ "$rpm_download_and_install" ]; then
             local pck2install_list=""
             for pck in ${SPEC_TARGET[@]}; do
-               filename="$rpms_dir/$target_cpu/\
+               if [ "${SPEC_BUILDARCH:-$BUILD_ARCH}" == "noarch" ]; then
+                  filename="$rpms_dir/noarch/\
+$pck-$pck_rpmversion-$pck_rpmrelease$DISTRO_rpm.noarch.rpm"
+               else
+                  filename="$rpms_dir/$target_cpu/\
 $pck-$pck_rpmversion-$pck_rpmrelease$DISTRO_rpm.$target_cpu.rpm"
+               fi
 
                [[ -n "$rpm_install_noinstall_mask" &&
                   $filename =~ $rpm_install_noinstall_mask ]] && continue
