diff -Nru autospec-1.5.0.orig//plugins/pck-update autospec-1.5.0/plugins/pck-update
--- autospec-1.5.0.orig//plugins/pck-update.in	2010-08-08 17:45:44.896183449 +0200
+++ autospec-1.5.0/plugins/pck-update.in	2010-08-08 19:43:32.147316080 +0200
@@ -436,9 +436,9 @@
            2>/dev/null >> $specfile_preprocessed" ||
         notify.error $"rpm exited with error code \`$?'"
 
-        SPECFILE_PREPROCESSED="$specfile_preprocessed"
-        notify.debug "SPECFILE_PREPROCESSED = \"$SPECFILE_PREPROCESSED\""
      fi
+     SPECFILE_PREPROCESSED="$specfile_preprocessed"
+     notify.debug "SPECFILE_PREPROCESSED = \"$SPECFILE_PREPROCESSED\""
      echo "SPECFILE_PREPROCESSED=\"$SPECFILE_PREPROCESSED\"" >> $infofile
 
      # set the variables 'rpmvar_name[]', 'rpmvar_value[]', 'rpmvar_expr[]'
