--- libapse.lib.in	2009-02-28 19:34:55.000000000 +0100
+++ libapse.lib.in.new	2009-03-02 18:46:02.000000000 +0100
@@ -403,7 +403,7 @@
 s/\.[^0-9].*//          # remove trailing stuff (.tar.gz, ...)
 s/.*-\([0-9]*.*\)/\1/   # <pck_name>-<pck_ver>
 s/.*_\([0-9]*.*\)/\1/   # <pck_name>_<pck_ver>
-s/.+\([0-9]\..*\)/\1/g  # <pck_name><pck_ver>" | \
+s/[^0-9]*\([0-9]\..*\)/\1/g  # <pck_name><pck_ver> | \
 sort -t. -n -r -k1 -k2 -k3 -k4 -k5 -k6 -k7 -k8 -k9 -k10 | head -n1)
       fi
 
