--- libapse.lib.in.2	2009-02-28 19:34:40.000000000 +0100
+++ libapse.lib.in	2009-02-28 19:34:55.000000000 +0100
@@ -392,7 +392,8 @@
 
       local pcknameregexp=$(echo $pcknameverbatim | sed "\
             s,%[{]*name[}]*,$specname,
-            s,%{.*}*,[^[:space:]-]*,")
+            s,%[{]*version[}]*,[A-Za-z0-9._]*,
+            s,%[{]*.*[}]*,[^[:space:]-]*,")
 
       if [ $? -eq 0 ]; then
          new_version=$(\
